serialization/json_value.go

```diff
@@ -16,8 +16,32 @@
 
 package serialization
 
+import "errors"
+
 type JSON []byte
 
 func (j JSON) String() string {
 	return string(j)
 }
+
+// MarshalJSON is used by functions in the Go `encoding/json` package. This
+// function ensures that a `serialization.JSON` value is correctly serialized
+// by this package. This is mostly applicable when `serialization.JSON` is
+// serialized as a field of a larger struct.
+func (j JSON) MarshalJSON() ([]byte, error) {
+	if j == nil {
+		return []byte("null"), nil
+	}
+	return j, nil
+}
+
+// UnmarshalJSON is used by functions in the Go `encoding/json` package. This
+// function ensures that a `serialization.JSON` value is correctly deserialized
+// from JSON objects that contain a field of this type.
+func (j *JSON) UnmarshalJSON(data []byte) error {
+	if j == nil {
+		return errors.New("serialization.JSON: UnmarshalJSON on nil pointer")
```

Comment:
Could you change this line to return a serialization error, like:
```
return ihzerrors.NewSerializationError("serialization.JSON: UnmarshalJSON on nil pointer", nil)
```
The corresponding import statement would be:
```
import ihzerrors "github.com/hazelcast/hazelcast-go-client/internal/hzerrors"
```

Reply:
^ It's even better to return an illegal argument error.

Reply:
Fixed in ab9de482c6c8c3a68a0c468890c7022da27ea8f6